s2s.go

```diff
@@ -20,17 +22,24 @@ func S2SClient(cfg *S2SClientConfig) *http.Client {
 	httpClient := &http.Client{
 		Transport: transport.Chain(http.DefaultTransport,
 			traceid.Transport,
-			transport.SetHeaderFunc("Authorization", func(req *http.Request) string {
-				return "BEARER " + S2SToken(cfg.JWTSecret, map[string]any{"service": cfg.Service})
-			}),
-			transport.If(cfg.DebugRequests, transport.LogRequests(transport.LogOptions{Concise: true, CURL: true})),
+			transport.SetHeader("User-Agent", fmt.Sprintf("sequence/%s", cfg.Service)),
+			transport.If(cfg.JWTSecret != "",
```

Comment:
We can also check the service name
```suggestion
			transport.If(cfg.JWTSecret != "" && cfg.Service != "",
```